tests/unit/admin/test-class-wc-rest-payments-orders-controller.php

```diff
@@ -141,14 +141,173 @@ public function test_capture_terminal_payment_success() {
 		$this->assertStringEndsWith( '/wc/v3/payments/readers/receipts/pi_mock', $result_order->get_meta( 'receipt_url' ) );
 	}
 
-	public function test_capture_terminal_payment_intent_non_capturable() {
+	public function test_capture_terminal_payment_succeeded_intent() {
+		$order = $this->create_mock_order();
+
+		$mock_intent = $this->createMock( WC_Payments_API_Intention::class );
+		$mock_intent
```

Comment:
`$mock_intent->expects( $this->any() )-> ...` calls are not making much sense, hence I'd only keep those for `get_id`, `get_status`:

```php
		$mock_intent = $this->createMock( WC_Payments_API_Intention::class );
		$mock_intent
			->expects( $this->atLeastOnce() )
			->method( 'get_status' )
			->willReturn( 'succeeded' );
		$mock_intent
			->expects( $this->atLeastOnce() )
			->method( 'get_id' )
			->willReturn( $this->mock_intent_id );

```

Reply:
Actually, these expectations are required to ensure mock values are returned correctly in the target code. Without them, the tests error out.

In any case, I've refined the `expects` for these mocks in c6a46d7.